src/staking/observable/index.ts

```diff
@@ -0,0 +1,372 @@
+import { networks, Psbt, Transaction } from "bitcoinjs-lib";
+import { ObservableStakingParams } from "../../types/params";
+import { UTXO } from "../../types/UTXO";
+import { StakingScripts } from "../stakingScript";
+import { StakingScriptData } from "../stakingScript";
+import { StakingError, StakingErrorCode } from "../../error";
+import { 
+  stakingTransaction, unbondingTransaction,
+  withdrawEarlyUnbondedTransaction,
+  withdrawTimelockUnbondedTransaction
+} from "..";
+import { 
+  getPublicKeyNoCoord, isTaproot,
+  isValidBitcoinAddress, isValidNoCordPublicKey
+} from "../../utils/btc";
+import { getStakingTerm, validateParams, validateStakingTxInputData } from "../../utils/staking";
+import { PsbtTransactionResult } from "../../types/transaction";
+
+interface StakerInfo {
+  address: string;
+  publicKeyNoCoordHex: string;
+}
+
+interface ObservableStakingTransaction {
+  txHex: string;
+  outputIndex: number;
```

Comment:
tbh I would rename this `stakingOutptuIndex`, as otherwise it may be about any output in staking transaction